### PR TITLE
Cleanup SnapshotsInProgress for Readability

### PR DESCRIPTION
**Pull request by original-brownbear**
* Formatting `SnapshotsInProgress` in a more reasonable way instead of mixing nested class, fields and methods.
* Removing pointless constants for the x-content serialization.
* Cleaning up double wrapping in `unmodifiableList`.
* Removing unused and incorrectly documented constructor from `SnapshotDeletionsInProgress`

no functional changes otherwise.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java [hunk 1]
```diff
@@ -105,795 +167,706 @@ public static Entry startedEntry(Snapshot snapshot, boolean includeGlobalState,
...
+                    }
+                }
+                builder.endArray();
+            }
+            builder.array("data_streams", dataStreams.toArray(new String[0]));
+            builder.endObject();
+            return builder;
+        }
+
+        private void addShardId(XContentBuilder builder, ToXContent indexId, int shardId, ShardSnapshotStatus status) throws IOException {
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java:835]: `addShardId` doesn't seem quite right, maybe something like `writeShardSnapshotStatus` instead?

**Review by DaveCTurner — APPROVED** [review]
Optional suggestion for a different name for the new method, otherwise LGTM

File: server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java (hunk 1 above)
Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java:835]: ++

**Comment by original-brownbear** [comment]
Thanks David!

**Pull request merged**